go/private/binary.bzl

```diff
@@ -164,8 +158,7 @@ def _extract_extldflags(gc_linkopts, extldflags):
       filtered_gc_linkopts += [opt]
   return filtered_gc_linkopts, extldflags
 
-def emit_go_link_action(ctx, transitive_go_library_paths, transitive_go_libraries, cgo_deps, libs,
-                         executable, gc_linkopts, x_defs):
+def emit_go_link_action(ctx, library, mode, executable, gc_linkopts, x_defs):
```

Comment:
Could you add a doc string to this function? I don't think the purpose of `mode` is obvious from the implementation of this function, or from a lot of the call sites, so it would be good to have some explanation.